packages/ra-core/src/form/useInput.ts

```diff
@@ -71,73 +84,68 @@ const useInput = ({
         ? composeValidators(validate)
         : validate;
 
-    // Fetch the initialValue from the record if available or apply the provided initialValue.
+    // Fetch the defaultValue from the record if available or apply the provided defaultValue.
     // This ensure dynamically added inputs have their value set correctly (ArrayInput for example).
-    // We don't do this for the form level initialValues so that it works as it should in final-form
-    // (ie. field level initialValue override form level initialValues for this field).
-    const { input, meta } = useFinalFormField(finalName, {
-        initialValue: get(record, source, initialValue),
-        defaultValue,
-        validate: sanitizedValidate,
-        ...options,
-    });
-
-    // Extract the event handlers so that we can provide ours
-    // allowing users to provide theirs without breaking the form
-    const { onBlur, onChange, onFocus, ...inputProps } = input;
-
-    const handleBlur = useCallback(
-        event => {
-            onBlur(event);
-            if (typeof customOnBlur === 'function') {
-                customOnBlur(event);
-            }
-        },
-        [onBlur, customOnBlur]
-    );
+    // We don't do this for the form level defaultValues so that it works as it should in react-hook-form
+    // (ie. field level defaultValue override form level defaultValues for this field).
+    const { field: controllerField, fieldState, formState } = useController({
+        name: finalName,
+        defaultValue: get(record, source, defaultValue),
+        rules: {
+            validate: async value => {
+                if (!sanitizedValidate) return true;
+                const error = await sanitizedValidate(
+                    value,
+                    formContext.getValues(),
+                    props
+                );
 
-    const handleChange = useCallback(
-        event => {
-            onChange(event);
-            if (typeof customOnChange === 'function') {
-                customOnChange(event);
-            }
+                if (!error) return true;
+                return getValidationErrorMessage(error);
+            },
         },
-        [onChange, customOnChange]
-    );
+        ...options,
+    });
 
-    const handleFocus = useCallback(
-        event => {
-            onFocus(event);
-            if (typeof customOnFocus === 'function') {
-                customOnFocus(event);
-            }
-        },
-        [onFocus, customOnFocus]
-    );
+    useApplyInputDefaultValues(props);
```

Comment:
There is something I don't understant (and that would merit a comment): you're already setting the defaultValue L93, why do you need to apply it again?

Reply:
Added a lengthy comment